File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        Anagrams = {}
        
        for string in strs:
            sortedString = ''.join(sorted(string))
            print(sortedString)
            
            if sortedString in Anagrams:
                Anagrams[sortedString].append(string)
            
            else: Anagrams[sortedString] = [string]
        
        return list(Anagrams.values())

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    hashTable = new Map()
    
    for(let i = 0; i <= strs.length - 1; i++){
        sortedString = strs[i].split('').sort().join('')
        
        //console.log(strs[i])
        //console.log(sortedString)
        //console.log(hashTable)
        
        if (hashTable[sortedString]) {
            hashTable[sortedString].push(strs[i])
        } else {
            hashTable[sortedString] = [strs[i]]
        }
        
        
        
    }
    
    return Object.values(hashTable)
    
};